internal/controllers/addonmetadata/controller.go

```diff
@@ -22,41 +22,49 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/log"
 
+	"github.com/go-logr/logr"
 	addonsv1alpha1 "github.com/mt-sre/addon-metadata-operator/api/v1alpha1"
 )
 
 // AddonMetadataReconciler reconciles a AddonMetadata object
 type AddonMetadataReconciler struct {
 	client.Client
+	Log    logr.Logger
 	Scheme *runtime.Scheme
 }
 
-//+kubebuilder:rbac:groups=addons.managed.openshift.io,resources=AddonMetadata,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=addons.managed.openshift.io,resources=AddonMetadata/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=addons.managed.openshift.io,resources=AddonMetadata/finalizers,verbs=update
-
-// Reconcile is part of the main kubernetes reconciliation loop which aims to
-// move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the AddonMetadata object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
-func (r *AddonMetadataReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
-
-	// your logic here
+//+kubebuilder:rbac:groups=addonsflow.redhat.openshift.io,resources=AddonMetadata,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=addonsflow.redhat.openshift.io,resources=AddonMetadata/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=addonsflow.redhat.openshift.io,resources=AddonMetadata/finalizers,verbs=update
 
-	return ctrl.Result{}, nil
-}
-
-// SetupWithManager sets up the controller with the Manager.
 func (r *AddonMetadataReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&addonsv1alpha1.AddonMetadata{}).
 		Complete(r)
 }
+
+// Reconcile - Implements the AddonMetadata flow
+// 1. Find corresponding imageset
+// 2. Validate the workload (extract bundles and index image)
+// 3. POST payloads to OCM
+func (r *AddonMetadataReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("addonmetadata", req.NamespacedName.String())
+
+	meta := &addonsv1alpha1.AddonMetadata{}
+	if err := r.Get(ctx, req.NamespacedName, meta); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	data, err := meta.ToJSON()
+	if err != nil {
+		log.Error(err, "could not marshal meta CR to JSON")
+	}
+	log.Info(string(data))
```

Comment:
we dont want to log for every reconcile loop

Reply:
this is just dummy code for now, will be wiped out by another PR. Basically this PR is committing all the required scaffolding, after which we can start focusing on reconciliation logic.